webapp/packages/core-blocks/src/CommonDialog/CommonDialog/CommonDialogWrapper.tsx

```diff
@@ -24,50 +23,35 @@ export interface CommonDialogWrapperProps {
   fixedSize?: boolean;
   fixedWidth?: boolean;
   freeHeight?: boolean;
-  autofocus?: boolean;
   className?: string;
   children?: React.ReactNode;
 }
 
 export const CommonDialogWrapper = observer<CommonDialogWrapperProps, HTMLDivElement>(
-  forwardRef(function CommonDialogWrapper(
-    { size = 'medium', fixedSize, fixedWidth, freeHeight, autofocus = true, 'aria-label': ariaLabel, className, children },
-    ref,
-  ) {
-    const [focusedRef] = useFocus({ autofocus });
+  forwardRef(function CommonDialogWrapper({ size = 'medium', fixedSize, fixedWidth, freeHeight, 'aria-label': ariaLabel, className, children }, ref) {
     const computedStyles = useS(styles);
     const context = useContext(DialogContext);
-    const dialogState = useDialogState({ visible: true });
 
-    useEffect(() => {
-      if (!dialogState.visible && !context.dialog.options?.persistent) {
+    function handleClose() {
+      if (!context.dialog.options?.persistent) {
         context.reject();
       }
-    });
+    }
 
     return (
       <Dialog
-        {...dialogState}
         ref={ref}
         aria-label={ariaLabel}
-        className={s(computedStyles, { container: true })}
-        visible={context.visible}
-        hideOnClickOutside={false}
-        modal={false}
+        open={context.visible}
+        data-size={size}
+        className={s(computedStyles, { dialog: true, fixedSize, fixedWidth, freeHeight }, className)}
+        backdrop
+        unmountOnHide
```

Comment:
`unmountOnHide` might break local state in complex dialogs with subdialogs, for example, an authentication dialog. If you encounter an error and click to show details, when you return, the local state of the authentication dialog will be lost.